docs/contracts/transient-storage.rst

```diff
@@ -0,0 +1,166 @@
+
+.. code-block:: solidity
+
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity ^0.8.27;
+
+    contract Generosity {
+        mapping(address => bool) sentGifts;
+        bool transient locked;
+
+        modifier nonReentrant {
+            require(!locked, "Reentrancy attempt");
+            locked = true;
+            _;
+            // Unlocks the guard, making the pattern composable.
+            // After the function exits, it can be called again, even in the same transaction.
+            locked = false;
+        }
+
+        function claimGift() nonReentrant public {
+            require(address(this).balance >= 1 ether);
+            require(!sentGifts[msg.sender]);
+            (bool success, ) = msg.sender.call{value: 1 ether}("");
+            require(success);
+
+            // In a reentrant function, doing this last would open up the vulnerability
+            sentGifts[msg.sender] = true;
+        }
+    }
+
+Transient storage is private to the contract that owns it, in the same way as persistent storage.
+Only owning contract frames may access their transient storage.
+And when they do, all the frames access the same transient store.
+
+Transient storage is part of the EVM state and is subject to the same mutability enforcements
+as persistent storage. As such, any read access to it is not ``pure`` and writing access is not ``view``.
+
+If the ``TSTORE`` opcode is called within the context of a ``STATICCALL``,
+it will result in an exception instead of performing the modification.
+``TLOAD`` is allowed within the context of a ``STATICCALL``.
+
+When transient storage is used in the context of ``DELEGATECALL`` or ``CALLCODE``,
+then the owning contract of the transient storage is the contract that issued ``DELEGATECALL``
+or ``CALLCODE`` instruction (the caller) as with persistent storage.
+When transient storage is used in the context of ``CALL`` or ``STATICCALL``,
+then the owning contract of the transient storage is the contract that is the target
+of the ``CALL``or ``STATICCALL`` instruction (the callee).
+
+.. note::
+    In the case of ``DELEGATECALL``, since references to transient storage variables
+    are currently not supported, it is not possible to pass those into library calls.
+    In libraries, access to transient storage is only possible using inline assembly.
+
+If a frame reverts, all writes to transient storage that took place between entry
+to the frame and the return are reverted, including those that took place in inner calls.
+The caller of an external call may bypass that mechanism using a ``try ... catch`` block.
+This mimics the behavior of persistent storage.
+
+.. note::
+    Currently, the compiler can parse ``transient`` as a data location, however it is not
+    defined as a keyword of the language yet. This means that the use of ``transient``
+    is backwards-compatible and does not break previous code that eventually used it as an identifier.
```

Comment:
I'd just drop that, it's more of an implementation detail than relevant information - or, if we want to keep it, I'd make it clear that it'll change with the next breaking release.